### PR TITLE
Google licenseclassifier pipeline

### PR DESCRIPTION
**Pull request by tdruez**
Creating a PR to simply the review/feedback process

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (tdruez) [inline .vscode/settings.json:1]: You need to add this file to your local gitignore.

File: docs/index.rst [hunk 2]
```diff
@@ -1,14 +1,9 @@
...
 
-Welcome to the very start of your ScanCode.io journey! In this documentation
-you’ll find information on:
-
-- An overview of ScanCode.io and ScanPipe
-- Installation instructions
-- Tutorials to get you started
-- Reference documentation about the ScanPipe concepts, Pipelines, Pipes and more
-- How to make technical contributions to the project and the community
+ScanCode.io is a server to script and automate software composition analysis
```
Comment (tdruez) [inline docs/index.rst:4]: Those changes have nothing to do here. There's a need for rebasing/merging the latest main.

File: docs/introduction.rst [hunk 3]
```diff
@@ -1,84 +1,77 @@
...
-
-**ScanCode.io** is a server to script and automate the process of
-**Software Composition Analysis (SCA)** to identify any open source components
-and their license compliance data in an application’s codebase. ScanCode.io can be
-used for various use cases, such as Docker container and VM composition
-analyses, among other applications.
-
-Why ScanCode.io?
-----------------
+Why ScanCode.io
```
Comment (tdruez) [inline docs/introduction.rst:3]: Same as above.

File: scanpipe/pipelines/docker.py [hunk 4]
```diff
@@ -75,7 +75,8 @@ def collect_images_information(self):
         Collect images information and store on project.
         """
         images_data = [docker.get_image_data(image) for image in self.images]
-        self.project.update_extra_data({"images": images_data})
+        self.project.extra_data.update({"images": images_data})
```
Comment (tdruez) [inline scanpipe/pipelines/docker.py:78]: Same as above.

File: scanpipe/pipelines/glc_scan.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
+from scanpipe.pipelines import Pipeline
+from scanpipe.pipes import glc
+from scanpipe.pipes import output
+from scanpipe.pipes import scancode
+from scanpipe.pipes.input import copy_inputs
+
+
+class TestPipeline(Pipeline):
```
Comment (tdruez) [inline scanpipe/pipelines/glc_scan.py:8]: We need a better name ;)

File: scanpipe/pipelines/glc_scan.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        scan_data = glc.to_dict(str(self.scan_output))
+        glc.create_codebase_resources(project, scan_data)
+
+    def csv_output(self):
+        """
+        Generate csv outputs.
+        """
+        output.to_csv(self.project)
+
+    steps = (
```
Comment (tdruez) [inline scanpipe/pipelines/glc_scan.py:64]: To remove the code duplication, you can now inherit from another Pipeline class such as ScanCodebase and define your steps as a classmethod, such as https://github.com/nexB/scancode.io/blob/main/scanpipe/pipelines/scan_codebase.py#L39 so you can remove the duplicated code from your class.

Comment (tdruez) [reply scanpipe/pipelines/glc_scan.py:64]: See https://github.com/nexB/scancode.io/blob/main/scanpipe/pipelines/docker.py#L28 as an example.

File: scanpipe/pipelines/root_filesystems.py [hunk 7]
```diff
@@ -56,15 +56,14 @@ def extract_input_files_to_codebase_directory(self):
         """
         input_files = self.project.inputs("*")
         target_path = self.project.codebase_path
-        errors = []
+        extract_errors = []
```
Comment (tdruez) [inline scanpipe/pipelines/root_filesystems.py:59]: Same as above.

File: scanpipe/pipelines/glc_scan.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
+from scanpipe.pipelines import Pipeline
```
Comment (tdruez) [inline scanpipe/pipelines/glc_scan.py:1]: The file header is missing.

File: scanpipe/pipes/glc.py [hunk 9]
```diff
@@ -0,0 +1,77 @@
+import io
```
Comment (tdruez) [inline scanpipe/pipes/glc.py:1]: The file header is missing.

File: scanpipe/tests/test_models.py [hunk 10]
```diff
@@ -937,26 +937,6 @@ def test_scanpipe_project_model_add_error(self):
         self.assertEqual("Error message", error.message)
         self.assertEqual("", error.traceback)
```
Comment (tdruez) [inline scanpipe/tests/test_models.py:938]: Same as above.

**Review by tdruez — COMMENTED** [review]
@AvishrantsSh See my comments for improvements. 
There's a need for cleanup, a lot of changes have nothing to do here.
Also, could you provide some help on how to install and run this pipeline?

File: docs/index.rst (hunk 2 above)
Comment (AvishrantsSh) [reply docs/index.rst:4]: I am not able to resolve this issue :( Merge and Rebase says Already up to date.
Also there are different versions of same file in the two branches but its not appearing during merge. 
How do I fix it?

Comment (AvishrantsSh) [reply docs/index.rst:4]: Maybe a fresh start? Because I've changed just a file or two.

Comment (AvishrantsSh) [reply docs/index.rst:4]: @tdruez I think I messed up a bit while updating branch

Comment (tdruez) [reply docs/index.rst:4]: @AvishrantsSh create a new branch based on latest `main` and commit your new files there.

**Pull request closed without merging**